test/uk/gov/hmrc/eventhub/config/TopicSpec.scala

```diff
@@ -116,9 +117,33 @@ class TopicSpec extends AnyFlatSpec with Matchers {
          |topics.email.${subscriber.name}.max-retries=0
          |""".stripMargin)
 
-    Topic
-      .configLoader(subscriptionDefaults)
-      .load(config, "topics") shouldBe Set(Topic("email", List(subscriber.copy(maxRetries = 0))))
+    val topic = Topic.configLoader(subscriptionDefaults).load(config, "topics").head
+
+    topic.name shouldBe "email"
+    val topicSubscriber = topic.subscribers.head
+    topicSubscriber.name shouldBe "foo-subscriber"
+    topicSubscriber.uri shouldBe Uri("http://localhost:8080/foo")
+    topicSubscriber.maxRetries shouldBe 0
+    topicSubscriber.maxBackOff shouldBe 2.seconds
+    topicSubscriber.minBackOff shouldBe 1.seconds
+  }
+
+  it should "load a topic with a subscriber that has path defined" in {
+    val config: Config = ConfigFactory.parseString(s"""
+                                                      |topics.email.${subscriber.name}.uri="${subscriber.uri}"
+                                                      |topics.email.${subscriber.name}.path="testPath"
+                                                      |""".stripMargin)
+
+    config.getValue("topics.email.foo-subscriber.path").toString should include("testPath")
+  }
+
+  it should "load a topic with a subscriber that has path None if its not defined" in {
+    val config: Config = ConfigFactory.parseString(s"""
+                                                      |topics.email.${subscriber.name}.uri="${subscriber.uri}"
+                                                      |topics.email.${subscriber.name}.path="${subscriber.pathFilter}"
+                                                      |""".stripMargin)
+
+    config.getValue("topics.email.foo-subscriber.path").leftSide.toString should include("None")
```

Comment:
these tests are only checking that the type safe config is loaded and does not check the parsing of the values into the `Subscriber `case class.
If the json path is an object and cannot be matched in the usual way then, perhaps do:
  ```
  val topic = Topic.configLoader(subscriptionDefaults).load(config, "topics").head

    topic.name shouldBe "email"
    val topicSubscriber = topic.subscribers.head
    topicSubscriber.pathFilter.toString ? shouldBe subscriber.pathFilter.toString
```

Reply:
Not sure if toString works in this way but it might be an option

Reply:
I am not using subscriptionDefaults

Reply:
I know, but you should be checking that config value has been parsed properly it's json path representation - this test does not check this.

Reply:
Oh, I see you are testing the config reader here! Never mind ignore me :)